stat/distuv/triangle.go

```diff
@@ -0,0 +1,192 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+)
+
+// Triangle represents a triangle distribution (https://en.wikipedia.org/wiki/Triangular_distribution).
+type Triangle struct {
+	a, b, c float64
+	Source  *rand.Rand
+}
+
+// Triangle constructs a new triangle distribution with lower limit a, upper limit b, and mode c.
+// Constraints are a < b and a ≤ c ≤ b.
+// This distribution is uncommon in nature, but may be useful for simulation.
+func NewTriangle(a, b, c float64) Triangle {
```

Comment:
I reviewed the code and all the non-sensical instances of divide by a-c (or similar) are unreachable in the case of c=a. Since it's a bit tedious to trace through, I've added a test case of c=a. It's also OK to set b=c; however, that test case would need a slightly looser tolerance on checkProbContinuous to pass, and I didn't think it was worth the code changes to prove that b=c is OK on every build.